packages/circuit-ui/components/List/List.tsx

```diff
@@ -96,6 +96,9 @@ const marginStyles = ({ noMargin }: ListProps) => {
   return css`
     label: list--no-margin;
     margin-bottom: 0;
+    li:last-child {
+      margin-bottom: 0;
+    }
```

Comment:
@connor-baer WDYT? The rest are smaller visual improvements in the stories, after you review this part I think we can merge this 🙂

Reply:
How does this affect nested lists?

Reply:
Do you mean like [this](https://oss-circuit-ui-ai0uf0bh1-sumup.vercel.app/?path=/story/typography-list--nested)? I think it's fine

Reply:
The problem is when there's another list item after the nested list:

![Screenshot 2021-07-13 at 18 23 07](https://user-images.githubusercontent.com/11017722/125489236-b1a697c6-afcc-4d55-9e9f-bcfbbca5244a.png)

Reply:
👀

Reply:
Nice catch, but this is a bug on `main` 🐛 

If you add `noMargin` to the nested list (which is the expected behavior, fixed the story) it would look like this:

<img width="399" alt="Screenshot 2021-07-13 at 18 57 35" src="https://user-images.githubusercontent.com/35560568/125494205-fd114148-faa1-4bda-885c-71f178292571.png">

- there is indeed an issue in the nested list because it doesn't have the 8px spacing with the next item 🐛 
- the missing spacing with the paragraph below the list is what we want, we can style it with the spacing mixin

An easy alternative would be _not_ to reset the outer spacing:

<img width="399" alt="Screenshot 2021-07-13 at 19 00 43" src="https://user-images.githubusercontent.com/35560568/125494684-0f5dc94e-a463-4a19-aa46-190bc09f8703.png">

- the list itself looks good
- but there is outer spacing, we don't want this optimally (though we had it before even with noMargin passed)

We can also address the actual bug but it looks pretty tricky to me, because the nested list isn't aware of any parent. Do you have a CSS magic trick for this. or should we keep it for a 3.x.x bugfix?

Reply:
In the future, we can use the `:not()` selector for this: https://twitter.com/leaverou/status/1354760561087676416. Right now, I'm not sure. 

This is an edge case and since this is an existing bug, let's address this in a separate PR 👍🏻

Reply:
This wasn't CSS magic, it was CSS 101 🤦 (I blame Tuesdays)

6fcc26c

Reply:
Ah but no, this also doesn't work since if a list _ends_ with a nested item, we'll be back at the unwanted outer margin

Reply:
I think I've got it now, we can reset a nested list's bottom margin when it's the `:last-child`

4ddb9b6

Reply:
You're the best 👏🏻